backend/src/chain/optimize.py

```diff
@@ -3,23 +3,54 @@
 from .chain import Chain
 
 
-def __removed_dead_nodes(chain: Chain) -> bool:
+class _Mutation:
+    def __init__(self) -> None:
+        self.changed = False
+
+    def signal(self) -> None:
+        self.changed = True
+
+
+def __removed_dead_nodes(chain: Chain, mutation: _Mutation):
     """
     If a node does not have side effects and has no downstream nodes, then it can be removed.
     """
-    changed = False
 
     for node in list(chain.nodes.values()):
         is_dead = len(chain.edges_from(node.id)) == 0 and not node.has_side_effects()
         if is_dead:
             chain.remove_node(node.id)
-            changed = True
+            mutation.signal()
             logger.debug(f"Chain optimization: Removed {node.schema_id} node {node.id}")
 
-    return changed
+
+def __static_switch_trim(chain: Chain, mutation: _Mutation):
+    """
+    If the selected variant of the Switch node is statically known, then we can remove the input edges of all other variants.
+    """
+
+    for node in list(chain.nodes.values()):
+        if node.schema_id == "chainner:utility:switch":
```

Comment:
in the interest of protecting my sanity, is there a way you could place this functionality in a lambda function defined on `@register` or something rather than doing it this way?

Reply:
Only if we make the backend representation of chains (+ the interface for optimizers) part of the API.

Reply:
Maybe then we just leave this as is for now and consider it in the future.